File: 0167_two_sum_ii_input_array_is_sorted.py
# https://leetcode-cn.com/problems/two-sum-ii-input-array-is-sorted/

class Solution:
    def twoSum(self, numbers, target):
        """
        Inputs:
            numbers: list[int]
            target: int
        Outputs:
            return list[int]
        """
        
        i = 0
        j = len(numbers) - 1
        while i < j:
            if numbers[i] + numbers[j] < target:
                i = i + 1
            elif numbers[i] + numbers[j] > target:
                j = j - 1
            else:
                break
        return [i+1,j+1]

File: Cpp/0167_two_sum_ii_input_array_is_sorted.cpp
// https://leetcode.cn/problems/two-sum-ii-input-array-is-sorted/

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        vector<int> res(2,0);
        while (left < right) {
            if (numbers[left] + numbers[right] > target) {
                right = right - 1;
            }
            else if (numbers[left] + numbers[right] < target) {
                left = left + 1;
            }
            else {
                res[0] = left + 1;
                res[1] = right + 1;
                return res;
            }
        }
        return res;
    }
};
